### PR TITLE
Add MASTG Components Instructions

### PR DESCRIPTION
**Pull request by cpholguera**
This pull request introduces comprehensive authoring and style guides for various MASTG documentation components. The guides establish standardized conventions for metadata, formatting, cross-linking, and writing across multiple documentation types to ensure consistency and clarity.

These guides will replace:
- [Writing MAS Weaknesses & Tests](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?usp=sharing)
- [Porting MASTG v1 Tests to v2](https://docs.google.com/document/d/1veyzE4cVTSnIsKB1DOPUSMhjXow_MtJOtgHeo5HVoho/edit?usp=sharing)

| File | Description |
| ---- | ----------- |
| porting-mastg-v1-tests-to-v2.instructions.md | Workflow and guidelines for migrating MASTG v1 tests to v2 format |
| mastg-tools.instructions.md | Standards for documenting tools used in tests, demos, and techniques |
| mastg-test.instructions.md | Comprehensive guide for authoring test documentation with required metadata and sections |
| mastg-techniques.instructions.md | Standards for documenting reusable procedures and workflows |
| mastg-rules.instructions.md | Guidelines for writing SAST rules, particularly for Semgrep |
| mastg-r2-scripts.instructions.md | Conventions for writing and using radare2 scripts in demos |
| mastg-mitmproxy-scripts.instructions.md | Standards for mitmproxy Python scripts in MASTG demos |
| mastg-knowledge.instructions.md | Guidelines for authoring knowledge articles explaining platform features |
| mastg-frida-scripts.instructions.md | Conventions for writing Frida scripts in demos |
| mastg-demo.instructions.md | Comprehensive guide for creating demonstrative examples with code samples |
| mastg-best-practice.instructions.md | Standards for documenting security best practices |
| mastg-apps.instructions.md | Guidelines for documenting reference applications |
| markdown.instructions.md | Global markdown formatting and writing style standards |

### CODE REVIEW

File: .github/instructions/mastg-test.instructions.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+- Static: “References to…” (semgrep/r2)
+- Dynamic: “Runtime Use …” (frida)
+
+Exceptions may apply where "Runtime ..." feels forced, for example tests using adb, local backups, or filesystem snapshots.
+
+Examples:
+
+- [MASTG-TEST-0207](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0207/)
+- [MASTG-TEST-0216](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0216/)
+- [MASTG-TEST-0263](https://mas.owasp.org/MASTG/tests-beta/android/MASVS-STORAGE/MASTG-TEST-0263/)
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md]: Maybe move them to the `## Tests` session as they are not "title-specific" and explain the reason i.e. `Tests to use as reference`

File: .github/instructions/mastg-test.instructions.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
+## Tests
+
+Tests are platform-specific and must be located under tests-beta/android/ or tests-beta/ios/, within the corresponding MASVS category. Their file names are the test IDs.
+
+Notes:
+
+- Tests with `platform: network` are still organized under the OS folder that the MASVS category belongs to (for example, Android network tests live under `tests-beta/android/MASVS-NETWORK/`).
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md]: Readability-wise, it might be smoother to read after the `ls -l` "structure" example. Wider context->edge case

File: .github/instructions/mastg-test.instructions.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+The MASWE weakness ID the test references.
+
+- In YAML front matter, specify the bare identifier (for example, `weakness: MASWE-0069`). In body text, include the leading `@` (for example, @MASWE-0069).
+
+#### type
+
+One or more test types.
+
+Supported: static, dynamic, network, manual, filesystem, developer.
+
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md:69]: Static & dynamic are self-explanatory but could be defined. The rest are not very obvious, so a small description might help the reader. Sample wording:

```suggestion

static: performed without executing the application (e.g by reading the source code)
dynamic: performed by executing the binary and testing it within a (closed) runtime environment
network: TODO
manual: tests that you have to use the app
filesystem: TODO
```

File: .github/instructions/mastg-test.instructions.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+Tests are platform-specific and must be located under tests-beta/android/ or tests-beta/ios/, within the corresponding MASVS category. Their file names are the test IDs.
+
+Notes:
+
+- Tests with `platform: network` are still organized under the OS folder that the MASVS category belongs to (for example, Android network tests live under `tests-beta/android/MASVS-NETWORK/`).
+
+Tests have an overview and contain Steps which produce outputs called observations. After following the Steps you come up with an [Observation](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.h9gqgz4hdubj) which you will [Evaluate](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.lare0v58fwbf).
+
+Example:
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md]: Maybe underline the reason why we have this example:

```suggestion
Example structure:
```

File: .github/instructions/mastg-test.instructions.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+This links to https://mas.owasp.org/MASTG/best-practices/MASTG-BEST-0001/
+
+Notes:
+
+- If no applicable best practices exist yet, you can omit the field or set an empty list: `best-practices: []`.
+
+#### prerequisites
+
+List prerequisites needed to execute or evaluate the test. Existing files are in prerequisites/. Create new ones if needed.
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md]:
```suggestion
List prerequisites needed to execute or evaluate the test. Existing files are in `prerequisites/`. Create new ones if needed.
```

File: .github/instructions/mastg-test.instructions.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+```md
+prerequisites:
+- identify-sensitive-data
+- identify-security-relevant-contexts
+```
+
+#### profiles
+
+Specify MASVS profiles where the test applies. Valid values: L1, L2, P, R.
+
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md:131]:
```suggestion
The profiles are described in [MAS Testing Profiles Guide]( https://docs.google.com/document/d/1paz7dxKXHzAC9MN7Mnln1JiZwBNyg7Gs364AJ6KudEs/edit?tab=t.0#heading=h.il6q80u4fm3n)

- L1 denotes Essential Security.
- L2 denotes Advanced Security.
```

File: .github/instructions/mastg-test.instructions.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+#### optional fields
+
+Include these if relevant:
+
+- `status:` draft, placeholder, deprecated
+- `note:` short free-form note
+- `available_since:` minimum platform/API level
+- `deprecated_since:` last applicable platform/API level
+
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md:152]:
```suggestion
`apis: TODO` 
```

File: .github/instructions/mastg-test.instructions.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+Tests have an overview and contain Steps which produce outputs called observations. After following the Steps you come up with an [Observation](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.h9gqgz4hdubj) which you will [Evaluate](https://docs.google.com/document/d/1EMsVdfrDBAu0gmjWAUEs60q-fWaOmDB5oecY9d9pOlg/edit?pli=1#heading=h.lare0v58fwbf).
+
+Example:
+
+```sh
+% ls -1 -F tests-beta/android/MASVS-CRYPTO/
+MASTG-TEST-0204.md
+MASTG-TEST-0205.md
+```
+
```
Comment (Diolor) [inline .github/instructions/mastg-test.instructions.md:28]: Add structural explanation and links to sessions:

```suggestion
Each test has two parts: the [Markdown metadata](#markdown-metadata) (YAML `front matter`) and the [Markdown body](#markdown-body).

```

File: .github/instructions/mastg-demo.instructions.md [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+Demos live in `demos/android/` or `demos/ios/` under the corresponding MASVS category folder. Each demo has its own folder named using its ID and contains:
+
+* Markdown file: `MASTG-DEMO-xxx.md`
+* Code samples (e.g. .kt, .swift, .xml, .plist)
+* Testing code (e.g. sh, py)
+* Output files (e.g. txt, json, sarif)
+
+**Language:** The samples are written in **Kotlin** or **Swift**, depending on the platform. In some cases, the samples will also include configuration files such as AndroidManifest.xml or Info.plist.
+
+**Decompiled Code:** If the sample can be decompiled, the decompiled code is also provided in the demo (e.g as a Java file on Android: `MastgTest_reversed.java`). This is useful for understanding the code in the context of the application.
```
Comment (Diolor) [inline .github/instructions/mastg-demo.instructions.md]: It would be helpful to add (or link to) the steps on how to create a `MastgTest_reversed.java` from the Kotlin file.

E.g. See `Run the Extraction Script` session in `MASTestApp` project.

Another idea might be helpful to add in this session, the maintainers' steps (indicative text): 
```
	1.	Open the MASTestApp for the respective platform (link placeholder). For additional information, refer to the README.md.
	2.	Implement the test.
	3.	Extract the required files (see section `Run the Extraction Script`).
	4.	Return the tests here in the final format.
```

File: .github/instructions/mastg-test.instructions.md
Comment (cpholguera) [inline .github/instructions/mastg-test.instructions.md:1]: Document prerequisites: Prerequisites are “organizational” things so to say. They are typically context that only the developer companies have such as what’s supposed to be sensitive data (and sensitivity levels), list of first-party packages, list of first-party network domains. An analysis (or automated testing tools) can make some assumptions and guesses but only the developer knows for real.

File: .github/instructions/mastg-test.instructions.md [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+One or more test types.
+
+Supported:
+
+- `static`: analysis of the app binary, reverse-engineered source code, or developer artifacts that are available in the APK/IPA app package (e.g., Android manifest, Info.plist, entitlements, etc.). No execution of the app is required.
+- `dynamic`: analysis of the app while it is running and involves runtime analysis such as hooking or method tracing.
+- `manual`: manual steps that require human judgment, such as inspecting app behavior, UI, or configuration. This may include reverse engineering or runtime analysis that cannot be fully automated.
+- `network`: analysis of network traffic, while the app is running. Done externally, for example, using a proxy or network capture tool.
+- `filesystem`: analysis of the app's file system, including local storage or backups, which doesn't involve runtime analysis such as hooking or method tracing.
+- `developer`: tests only the developer can perform because they require access to the source code, build process, or other internal resources.
```
Comment (cpholguera) [inline .github/instructions/mastg-test.instructions.md]: Suggestion by @TheDauntless 

```suggestion
- `source-code`: tests only the developer can perform because they require access to the source code, build process, or other internal resources.
```

File: .github/instructions/mastg-test.instructions.md [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+
+This links to https://mas.owasp.org/MASTG/best-practices/MASTG-BEST-0001/
+
+Notes:
+
+- If no applicable best practices exist yet, you can omit the field or set an empty list: `best-practices: []`.
+
+#### prerequisites
+
+List the prerequisites needed to execute or evaluate the test. Existing files are in `prerequisites/`. Create new ones if required.
```
Comment (cpholguera) [inline .github/instructions/mastg-test.instructions.md]:
```suggestion
List the conditions that must be known or available before running or evaluating the test. These items capture internal context that only the developer or the organization can provide. Existing files are in the `prerequisites/` folder. Create new ones when needed.

Common examples include:

- Defined categories of sensitive data and their sensitivity levels used within the app.
- A list of first party packages, libraries, and modules.
- A list of first party network domains and services the app is expected to contact.
```

File: .github/instructions/mastg-test.instructions.md [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+
+- If there are no prerequisites, you can omit this field or use an empty list.
```
Comment (cpholguera) [inline .github/instructions/mastg-test.instructions.md]:
```suggestion
If there are no prerequisites, you can omit this field or use an empty list.
```
